### PR TITLE
Pipeop hurdle [experimental] 

### PR DESCRIPTION
**Pull request by pfistfl**
**Update**
This PR currently exists in order to keep track of the code and issue.
MultiOutput / Zero-Inflation will be covered via `mlr3multioutput`, OVR and OVO multi-to-binary are already covered via `ppl("ovr")` 

# Current scope of the PR:

**New PipeOps**: 

- `PipeOpMutateTarget`: Works like `PipeOpMutate`, i.e. transforms data according to a formula.
  but works on the `target` column(s). During prediction, the target is set to `NA`.

- `PipeOpNewTarget`: Sets a new target column and changes the `task_type` accordingly.
  Currently, the user has to specify `new_target_name` and `task_type`. Old target is set 
  to a new col role: `unused`. The new target has to exist in the data. 

This enables:
- Zero-Inflated Models
- Multiple targets and multiple outputs (via chains)
- (future) 1-vs-all and 1-vs-one multiclass to binary task reduction.

Further Aspects:
- The multi-class to binary class reduction requires an additional `PipeOp` that can convert
   probabilities for binary tasks to a multiclass prediction. This is currently not trivial,
   as the input is a list of predictions for binary tasks, but those now lack the `class_levels`, 
   i.e. we might have to also pass the task (I guess mixed type inputs are currently not fully supported).

- In order for multi-output like chains to work, I needed a way to set some columns to unused, which is why I currently add this to `mlr_reflections`. We might need to discuss a way to do this!

**Usecase**
For the usecase, we assume we have 3 targets, y1 to y3. We want to predict y3 using y1, y2
What the following figure displays is basically: 
set a new target (y1) -> copy(2) -> (NULL, learnerCV) -> featunion(2) ->
  set a new target (y2) -> copy(2) -> (NULL, learnerCV) -> featunion(2) ->
  set a new target (y3) -> lrn

`copy(2) %>>% (NULL, learnerCV) %>>%  featunion(2)` is used in order to keep the original feature and cbind prediction for intermediate target. 

multiout

### CODE REVIEW

**Comment by pfistfl** [comment]
@mllg @berndbischl This could need a conceptual (not code, but ideas) review.

File: R/PipeOpNewTarget.R [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+#' )
+#'
+#' pom$train(list(tsk("iris")))[[1]]
+PipeOpNewTarget = R6Class("PipeOpNewTarget",
+  inherit = PipeOp,
+  public = list(
+    initialize = function(id = "new_target", param_vals = list()) {
+      ps = ParamSet$new(params = list(
+        ParamUty$new("new_target", tags = c("train", "predict", "required"),
+          custom_check = function(x) {check_string(x, null.ok = TRUE)}),
```
[the remark this answers is not in the document]
Comment (mb706) [reply R/PipeOpNewTarget.R]: check_character, length >= 1

File: R/PipeOpNewTarget.R [hunk 2]
```diff
@@ -0,0 +1,123 @@
+#' @title PipeOpNewTarget
+#' @name mlr_pipeops_newtarget
+#' @format Abstract [`R6Class`] inheriting from [`PipeOp`].
+#'
+#' @description
+#'   Substitutes the current target with a new target.
+#'   The new target must be an existing column in the Task.
+#'   The previous target is set to row_role: `unused'.
```
Comment (mb706) [inline R/PipeOpNewTarget.R]: would just remove the role

File: R/PipeOpNewTarget.R [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      private$convert_task_type(inputs)
+    }
+  ),
+  private = list(
+    convert_task_type = function(inputs) {
+      intask = inputs[[1]]$clone(deep = TRUE)
+      assert_choice(self$param_set$values$new_target, intask$col_info$id)
+      task = convert_task(intask, self$param_set$values$new_task_type, self$param_set$values$new_target)
+      if (all(self$param_set$values$new_target != intask$target_names))
+        task$set_col_role(intask$col_roles$target, "unused")
```
Comment (mb706) [inline R/PipeOpNewTarget.R]: set_col_role(... character(0))

File: R/PipeOpNewTarget.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (mb706) [reply R/PipeOpNewTarget.R]: use selector instead

**Comment by pfistfl** [comment]
Currently blocked by #357

**Comment by pfistfl** [comment]
Will not delete the branch for now as it might contain relevant code

**Pull request closed without merging**